src/rust/src/pkcs7.rs

```diff
@@ -262,6 +262,26 @@ fn sign_and_serialize<'p>(
     }
 }
 
+fn compute_pkcs7_signature_algorithm<'p>(
+    py: pyo3::Python<'p>,
+    private_key: &'p pyo3::PyAny,
+    hash_algorithm: &'p pyo3::PyAny,
+    rsa_padding: &'p pyo3::PyAny,
+) -> pyo3::PyResult<common::AlgorithmIdentifier<'static>> {
+    let key_type = x509::sign::identify_key_type(py, private_key)?;
+    let has_pss_padding = !rsa_padding.is_none() && rsa_padding.is_instance(types::PSS.get(py)?)?;
```

Comment:
```suggestion
    let has_pss_padding = rsa_padding.is_instance(types::PSS.get(py)?)?;
```

I don't think the `is_none()` check is required? None isn't an instance of PSS :-)

Reply:
Ah, I got that code from https://github.com/pyca/cryptography/blob/50ea0faab70d2830e7d89756731fecf9ca64528e/src/rust/src/x509/sign.rs#L135-L137. Should I remove it from there as well?

Reply:
Probably that should be fixed, but no need to do it in this PR.

Reply:
done